aws/biganimal-preflight-aws

```diff
@@ -0,0 +1,478 @@
+  echo -e "You can also run ${GREEN}aws support create-case help${NC} for more examples."
+
+  rm -rf "${TMPDIR}"
+}
+trap _onexit EXIT
+pushd "${TMPDIR}" > /dev/null 2>&1 || exit
+
+TMP_SUGGESTION=$TMPDIR/suggestions_$$
+touch "$TMP_SUGGESTION"
+
+function store_suggestion()
+{
+    echo "$1" >> "$TMP_SUGGESTION"
+}
+
+#### Check AWS CLI version
+function version { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
+function check_aws_version {
+  fullcliversion=$(aws --version)
+  cliversionstripped=${fullcliversion% Python*}
+  cliversion=${cliversionstripped#*/}
+  if [ "$AWS_EXECUTION_ENV" == "CloudShell" ]; then
+    suggest "Run AWS Preflight Checks with aws-cli ${cliversion} in AWS CloudShell" ok
+  else
+    [ "$(version "$cliversion")" -lt "$(version "2.3.0")" ] && suggest "error: upgrade aws-cli to 2.3.0 or later" alert && exit 1
+    suggest "Run AWS Preflight Checks with aws-cli ${cliversion}" ok
+  fi
+}
+check_aws_version
+
+function validate_account() {
+  # TODO
+  # check if AWS CLI has something to verify if an AWS Account is suspended
+  # was looking at aws organizations describe-account --account-id XXXX
+  # but I can't execute that without enough rights on the management account
+  store_suggestion "Make sure the AWS Account ID ${GREEN}$1${NC} is the one that you want to use for BigAnimal"
+}
+
+#### AWS User Role Assignment Checking
+function validate_role_assignment() {
+  # TODO
+  # check if AWS CLI has something to verify the IAM permissions assigned to the
+  # user/role/token that is being used to make the AWS API calls
+  # was looking at
+  # aws iam list-user-policies
+  # aws iam list-attached-user-policies
+  # aws iam list-groups-for-user
+  # For each group: aws iam list-group-policies and aws iam list-attached-group-policies
+  # For roles: aws iam list-role-policies and aws iam list-attached-role-policies
+  # but maybe this not make really sense because you need to have also IAM rights
+  # to query for IAM itself like GetUser, GetRole, GetPolicy that are not really
+  # needed to be able to use CreateRole, AttachRolePolicy, PutRolePermissionsBoundary,
+  # CreatePolicy, CreatePolicyVersion, UpdateAssumeRolePolicy
+  store_suggestion "Make sure the AWS credentials used ${GREEN}$1${NC} has rights to create IAM Roles and Policies"
+}
+
+function check_onboard() {
+  #### AWS User and Account Checking
+  current_aws_user=$(aws sts get-caller-identity --no-cli-pager --query 'UserId' --output text)
+  validate_account "$account"
+  validate_role_assignment "$current_aws_user"
+}
+
+[ "$onboard" = "true" ] && check_onboard
+
+echo ""
+echo "##############################################"
+echo -e "Checking Service Quotas Limits on ${GREEN}$region${NC}..."
+echo "##############################################"
+echo ""
+# getting resources current usage
+start_time=$(date -d "3 hours ago" '+%Y-%m-%dT%H:%M:%S')
```

Comment:
```suggestion
start_time=$(date -d "3 hours ago" '+%Y-%m-%dT%H:%M:%S')
```
this doesn't work on bash 5 on macos Monterey, what is the target platform?

Reply:
Good catch here! Worked to me because I have coreutils installed (with GNU date), I'll update the script accordingly. I was aiming to have it usable on all the major platforms:
Should run/be tested on
- MacOS Monterey with bash installed
- AWS CloudShell with Amazon Linux 2
- Ubuntu 20.04 
- Windows 10 with Ubuntu on WSL